elements/lisk-p2p/src/p2p_types.ts

```diff
@@ -13,24 +13,40 @@
  *
  */
 import { ConnectionKind, PeerKind } from './constants';
-export interface P2PPacket {
-	readonly data?: unknown;
-}
 
-export interface P2PRequestPacket extends P2PPacket {
+export interface P2PRequestPacket {
 	readonly data?: unknown;
 	readonly procedure: string;
 }
 
-export interface P2PResponsePacket extends P2PPacket {
+export interface P2PResponsePacket {
 	readonly data: unknown;
 }
 
-export interface P2PMessagePacket extends P2PPacket {
+export interface P2PMessagePacket {
```

Comment:
wouldn't it be easier to just include `peerId` in this packet?

Reply:
That is the first option that I've assessed as it is indicated in the issue. The problem is that P2PMessagePacket interface is used as well in the `send()` from peer pool. It doesn't make sense to include the peerId in the message because it would be redundant. When you receive a package from a peer, we are able to grab the peerId at that moment. There is no need to include this information in the package itself.

Reply:
I think it make sense to separate them if the `send()` is using it.
but for user, it would be nice to have similar response interface for send and request if above comment is handled